tools/data_gen/src/main.rs

```diff
@@ -39,6 +45,10 @@ struct FilePaths {
     /// library path
     #[argh(option, short = 'l', default = "Path::new(\".\").into()")]
     pub lib_path: PathBuf,
+
+    /// whether data is fixpoint or int
+    #[argh(option, short = 'f', default = "false")]
+    pub fp_data: bool,
```

Comment:
This should be information provided by the frontend program generating the Calyx program by adding some attributes like this:
```
@fp_width(32) @int_width(16) mem = std_mem_d1(...)
```

#315 also talks about this attribute-based typing layer. However, Dahlia does not emit attributes like this so for now we can merge this. However, we should open an issue saying that this is what we want to do in the future.